### PR TITLE
A4A > Agency Tier: implement feature announcement modal

### PR DESCRIPTION
**Pull request by yashwin**
## Proposed Changes

This PR implements a feature announcement modal for the Agency Tier.

## Why are these changes being made?

* To show a feature announcement modal for 

## Testing Instructions

* Use the agency MC tool > Set the agency tier to undefined
* Open the A4A live link > Go to the Overview page > Verify that a modal is displayed as shown below & do not dismiss the modal. Also, verify it looks good on mobile view.
* Use the MC tool to update the agency tier to "Emerging Partner" > Verify that a modal is displayed as shown below. Dismiss the modal and refresh the page > Verify that it is not shown again.
* Also perform the test instructions here: https://github.com/Automattic/wp-calypso/pull/96486 and see if it works perfectly.

Screenshot 2024-11-18 at 8 38 07 PM

Screenshot 2024-11-18 at 8 38 22 PM

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/constants/index.ts [hunk 1]
```diff
@@ -7,4 +7,4 @@ export const JETPACK_CONNECT_A4A_LINK =
 
 export const REFERRAL_EMAIL_QUERY_PARAM_KEY = 'referral_email';
 
-export const AGENCY_FIRST_PURCHASE_SESSION_STORAGE_KEY = 'a4a-first-purchase';
+export const AGENCY_FIRST_PURCHASE_SESSION_STORAGE_KEY = 'a4a_first_purchase';
```
Comment (yashwin) [inline client/a8c-for-agencies/constants/index.ts:10]: Updated the key to make sure we show the modal only after the feature is shipped.

File: client/a8c-for-agencies/sections/marketplace/products-overview/hooks/use-issue-and-assign-licenses.tsx [hunk 2]
```diff
@@ -133,11 +134,13 @@ function useIssueAndAssignLicenses(
...
-				// We need to show the agency tier celebration modal after the first purchase
-				sessionStorage.setItem( AGENCY_FIRST_PURCHASE_SESSION_STORAGE_KEY, 'true' );
-				dispatch( fetchAgencies() );
+			if ( isEnabled( 'a8c-for-agencies-agency-tier' ) ) {
+				// Refresh the list of agencies if we don't have the tier to get the updated tier
+				if ( ! agency?.tier ) {
+					// We need to show the agency tier celebration modal after the first purchase
+					sessionStorage.setItem( AGENCY_FIRST_PURCHASE_SESSION_STORAGE_KEY, 'true' );
+					dispatch( fetchAgencies() );
+				}
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/products-overview/hooks/use-issue-and-assign-licenses.tsx:143]: Set the session storage key and fetch agencies only if the feature is enabled

**Review by tiagonoronha — APPROVED** [review]
LGTM 👍

**Review by simonktnga8c — APPROVED** [review]
Looks great! 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17008025

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17008025).

**Pull request merged**